faunadb/client.go

```diff
@@ -215,33 +263,49 @@ func (client *FaunaClient) newClient(basicAuth string, observer ObserverCallback
 	return &FaunaClient{
 		basicAuth:        basicAuth,
 		endpoint:         client.endpoint,
+		headers:          client.headers,
 		http:             client.http,
 		isTxnTimeEnabled: client.isTxnTimeEnabled,
+		queryTimeoutMs:   client.queryTimeoutMs,
 		lastTxnTime:      client.lastTxnTime,
 		observer:         observer,
 	}
 }
 
-func (client *FaunaClient) performRequest(expr Expr) (response *http.Response, err error) {
-	var request *http.Request
+func (client *FaunaClient) performRequest(expr Expr, configs []QueryConfig) (response *http.Response, err error) {
 
-	if request, err = client.prepareRequest(expr); err == nil {
+	if request, err := client.prepareRequest(expr, configs); err == nil {
```

Comment:
The previous code was correct. If you replace `=` with `:=` you're redeclaring the `err` variable while `return` will still yield back the original `err` variable declared as a named return. Thus, if `prepareRequest` fail, you're not gonna return that error. Instead, the result will be `nil` for both `response` and `err`.